File: lib/models/amap_regeo_code.dart
import 'package:amap_flutter_search/models/amap_address_component.dart';
import 'package:amap_flutter_search/models/amap_aoi.dart';
import 'package:amap_flutter_search/models/amap_poi.dart';
import 'package:amap_flutter_search/models/amap_road.dart';
import 'package:amap_flutter_search/models/amap_road_inter.dart';
import 'package:json_annotation/json_annotation.dart';
part 'amap_regeo_code.g.dart';

@JsonSerializable()
class AMapReGeocode {
  ///格式化地址
  String? formattedAddress;

  ///地址组成要素
  AMapAddressComponent? addressComponent;

  ///道路信息 AMapRoad 数组
  List<AMapRoad?>? roads;

  ///道路路口信息 AMapRoadInter 数组
  List<AMapRoadInter?>? roadinters;

  ///兴趣点信息 AMapPOI 数组
  List<AMapPOI?>? pois;

  ///兴趣区域信息 AMapAOI 数组
  List<AMapAOI?>? aois;

  AMapReGeocode();

  factory AMapReGeocode.fromJson(Map<String, dynamic> srcJson) =>
      _$AMapReGeocodeFromJson(srcJson);

  Map<String, dynamic> toJson() => _$AMapReGeocodeToJson(this);
}

File: android/src/main/java/com/amap/amap_flutter_search/AMapReGeocode.java
package com.amap.amap_flutter_search;

import java.util.List;

public class AMapReGeocode {
    ///格式化地址
    private String formattedAddress;

    ///地址组成要素
    private AMapAddressComponent addressComponent;

    ///道路信息 AMapRoad 数组
    private List<AMapRoad> roads;

    ///道路路口信息 AMapRoadInter 数组
    private List<AMapRoadInter> roadinters;

    ///兴趣点信息 AMapPOI 数组
    private List<AMapPOI> pois;

    ///兴趣区域信息 AMapAOI 数组
    private List<AMapAOI> aois;

    public String getFormattedAddress() {
        return formattedAddress;
    }

    public void setFormattedAddress(String formattedAddress) {
        this.formattedAddress = formattedAddress;
    }

    public AMapAddressComponent getAddressComponent() {
        return addressComponent;
    }

    public void setAddressComponent(AMapAddressComponent addressComponent) {
        this.addressComponent = addressComponent;
    }

    public List<AMapRoad> getRoads() {
        return roads;
    }

    public void setRoads(List<AMapRoad> roads) {
        this.roads = roads;
    }

    public List<AMapRoadInter> getRoadinters() {
        return roadinters;
    }

    public void setRoadinters(List<AMapRoadInter> roadinters) {
        this.roadinters = roadinters;
    }

    public List<AMapPOI> getPois() {
        return pois;
    }

    public void setPois(List<AMapPOI> pois) {
        this.pois = pois;
    }

    public List<AMapAOI> getAois() {
        return aois;
    }

    public void setAois(List<AMapAOI> aois) {
        this.aois = aois;
    }
}
